neo.UnitTests/Consensus/UT_Consensus.cs

```diff
@@ -112,25 +109,185 @@ public void ConsensusService_Primary_Sends_PrepareRequest_After_OnStart()
+            Console.WriteLine("\nCN2 simulation time");
+            actorConsensus.Tell(getCommitPayloadModifiedAndSignedCopy(commitPayload, 1, kp_array[1], blockToSign));
+
+            Console.WriteLine("\nCN3 simulation time");
+            actorConsensus.Tell(getCommitPayloadModifiedAndSignedCopy(commitPayload, 2, kp_array[2], blockToSign));
+
+            Console.WriteLine("\nCN4 simulation time");
+            actorConsensus.Tell(getCommitPayloadModifiedAndSignedCopy(commitPayload, 3, kp_array[3], blockToSign));
+
+            // =============================================
+            // Testing commit with wrong signature not valid
+            // It will be invalid signature because we did not change ECPoint
+            Console.WriteLine("\nCN6 simulation time. Wrong signature, KeyPair is not known");
+            actorConsensus.Tell(getPayloadAndModifyValidator(commitPayload, 5));
+
+            Console.WriteLine("\nWaiting for recovery due to failed nodes... ");
+            var backupOnRecoveryMessageAfterCommit = subscriber.ExpectMsg<LocalNode.SendDirectly>();
+            var rmPayload = (ConsensusPayload)backupOnRecoveryMessageAfterCommit.Inventory;
+            RecoveryMessage rmm = (RecoveryMessage)rmPayload.ConsensusMessage;
+            // =============================================
+
+            Console.WriteLine("\nCN5 simulation time");
+            actorConsensus.Tell(getCommitPayloadModifiedAndSignedCopy(commitPayload, 4, kp_array[4], blockToSign));
+
+            var onBlockRelay = subscriber.ExpectMsg<LocalNode.Relay>();
+            var utBlock = (Block)onBlockRelay.Inventory;
+
+            //var lastPayloadOfChangeView = subscriber.ExpectMsg<Neo.Consensus.ConsensusService.Timer>();
+
+            Console.WriteLine($"\nAsserting block NextConsensus..{utBlock.NextConsensus}");
+            utBlock.NextConsensus.Should().Be(updatedContract.ScriptHash);
 
             // ============================================================================
             //                      finalize ConsensusService actor
             // ============================================================================
-
             //Thread.Sleep(4000);
+            Console.WriteLine("Finalizing consensus service actor and returning states.");
+
+            // Returning values to context beucase tests are not isolated
+            mockContext.Object.Block.NextConsensus = originalContract.ScriptHash;
+            mockContext.Object.PrevHeader.NextConsensus = originalContract.ScriptHash;
+            mockContext.Object.Block.MerkleRoot = originalBlockMerkleRoot;
+
+            mockContext.Object.Reset(0);
             Sys.Stop(actorConsensus);
             TimeProvider.ResetToDefault();
-
+            // Ensure thread is clear
             Assert.AreEqual(1, 1);
         }
 
+        public ConsensusPayload getCommitPayloadModifiedAndSignedCopy(ConsensusPayload cpToCopy, ushort vI, KeyPair kp, Block blockToSign)
```

Comment:
Hi @vncoelho, please use upper camel case to name methods.

Reply:
true.